src/boot/efi/util.h

```diff
@@ -74,4 +74,11 @@ static inline void FileHandleClosep(EFI_FILE_HANDLE *handle) {
 #define UINT64_MAX ((UINT64) -1)
 #endif
 
+VOID log_error_stall(const CHAR16 *fmt, ...);
 EFI_STATUS log_oom(void);
+
+#define log_error_status_stall(err, fmt, ...) \
+        ({ \
+                log_error_stall(fmt, ##__VA_ARGS__, err); \
```

Comment:
hmm, i'd probably not add the err implicitly to the end. That only works safely if the user's format string really ends in ": %r". But that sounds like dangerous bet.

i'd keep the macro, but would leave it to the caller to use both %r and add the final arg if they want.

Reply:
Providing err twice would kind of defeat the purpose of this macro...

Reply:
I still see benefit, as it means you can write stuff:

```c
return log_error_status_stall(x, …);
```
instead of:

```c
{
        log_error_stall(…);
        return x;
}
```

one line instead of four sounds like it's worth it, given we do this often.

Also, it's good to have this behave closely (if not the same) to how things are done in our userspace codebase.

Reply:
Yes, but in userspace you don't need to provide errno a second time because of %m, this macro is supposed to mirror that. Shouldn't a comment be enough here?

Reply:
yes, userspace semantics are different, %m there references errno unconditionally, while %r here expects a variable to be passed explicitly. I think we should acknowledge the difference and not hide it.

Or to say this differently,  I want that stuff like this is OK:

```c
log_error_stall(err, "Failed to frobnicate '%s': %r\n"
                     "(Please see https://frobnicator.com/explanation.html for details on debugging frobnication issues with '%s'.)",
                s, err, s);
```

Reply:
Comments are always good, i.e. pointing people to the fact that %r in efi expects an argument while %m in userspace next to this macro definitely makes sense.

Reply:
How about two macros instead?
return log_error_with_r_stall(EFI_INVALID_PARAMETER, L"Error: %r");
return log_error_status_stall(EFI_INVALID_PARAMETER, L"Error getting device path.", EFI_INVALID_PARAMETER);

Reply:
i ike the version currently in the PR, it's nice and short enough I think. I would just leave it as is now